doi-transformer/src/main/java/no/unit/nva/doi/transformer/CrossRefConverter.java

```diff
@@ -165,46 +177,120 @@ private Range extractPages(CrossRefDocument document) {
+        // TODO actually call the Channel Register API and get the relevant details
+        return new Journal.Builder()
+                .withLevel(null)
+                .withTitle(extractJournalTitle(document))
+                .withOnlineIssn(extractOnlineIssn(document))
+                .withPrintIssn(extractPrintIssn(document))
+                .withOpenAccess(false)
+                .withPeerReviewed(false)
+                .build();
+    }
+
+    private boolean hasReviews(CrossRefDocument document) {
+        return nonNull(document.getReview());
+    }
+
+    private List<String> extractIsbn(CrossRefDocument document) {
+        List<Isxn> isbns = document.getIsbnType();
+        if (isNull(isbns) || isbns.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return isbns.stream()
+                .map(Isxn::getValue)
+                .collect(Collectors.toList());
+
+    }
+
+    private String extractSeriesTitle(CrossRefDocument document) {
+        if (isNull(document.getContainerTitle()) || document.getContainerTitle().isEmpty()) {
+            return null;
+        }
+        return document.getContainerTitle().stream()
+                .findFirst()
+                .orElse(null);
+    }
+
+    private String extractPublisherName(CrossRefDocument document) {
+        return document.getPublisher();
+    }
+
+    private Organization extractAndCreatePublisher(CrossRefDocument document) {
+        if (isNotEmpty(document.getPublisher())) {
+            return new Organization.Builder().withLabels(Map.of("name", document.getPublisher())).build();
+        }
+        return null;
     }
 
     private String extractPrintIssn(CrossRefDocument document) {
-        return IssnCleaner.clean(filterIssnsByType(document, Issn.IssnType.PRINT));
+        return IssnCleaner.clean(filterIssnsByType(document, Isxn.IsxnType.PRINT));
     }
 
     private String extractOnlineIssn(CrossRefDocument document) {
-        return IssnCleaner.clean(filterIssnsByType(document, Issn.IssnType.ELECTRONIC));
+        return IssnCleaner.clean(filterIssnsByType(document, Isxn.IsxnType.ELECTRONIC));
     }
 
-    private String filterIssnsByType(CrossRefDocument crossRefDocument, Issn.IssnType type) {
-        List<Issn> issns = crossRefDocument.getIssnType();
+    private String filterIssnsByType(CrossRefDocument crossRefDocument, Isxn.IsxnType type) {
+        List<Isxn> issns = crossRefDocument.getIssnType();
         if (isNull(issns) || issns.isEmpty()) {
             return null;
         }
-
         return issns.stream().filter(issn -> issn.getType().equals(type))
-                .map(Issn::getValue)
+                .map(Isxn::getValue)
                 .findAny()
                 .orElse(null);
     }
 
     private PublicationInstance<?> extractPublicationInstance(CrossRefDocument document) {
+        switch (CrossrefType.getByType(document.getType())) {
```

Comment:
This seems very simillar to the code in line 183

Reply:
Almost, one is PublicationContext and one is PublicationInstance<> and it seems like both will be replaces in a future modeel....